pkg/runtime/reconciler.go

```diff
@@ -292,12 +288,13 @@ func (r *resourceReconciler) ensureConditions(
 	exit := rlog.Trace("r.ensureConditions")
 	defer exit(err)
 
-	if syncedCond := ackcondition.Synced(res); syncedCond == nil {
-		rlog.Debug("resource missing ACK.ResourceSynced condition")
-		// only the resource manager will know whether the resource is in a
-		// stable sync state. Even if we got no error back from the
-		// create/update operations, we can only set this to Unknown.
-		condStatus := corev1.ConditionUnknown
+	// If the ACK.ResourceSynced condition is not set using the custom hooks,
+	// determine the Synced condition using "rm.IsSynced" method
+	if ackcondition.Synced(res) == nil {
+		condStatus := corev1.ConditionFalse
+		if synced, _ := rm.IsSynced(ctx, res); synced {
```

Comment:
I know that generated `IsSynced` function returns only nil `err` for now. Still we should handle this error properly?

Reply:
I did think about it before sending PR and my thought process was that we want to mark `ResourceSynced` condition False when `rm.IsSynced` function returns false or non-nil error. And the code in PR did that.

Saying that I also agree to your point and I have updated the code to handle error from `rm.IsSynced` function. Following are controller logs for error vs no-error case now.

Error:
```
1.645553470691295e+09   DEBUG   ackrt   >> r.ensureConditions   {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}
1.6455534706913338e+09  DEBUG   ackrt   >>> rm.IsSynced {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}
1.645553470691346e+09   DEBUG   ackrt   <<< rm.IsSynced {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2, "error": "error from IsSynced"}
1.645553470691356e+09   DEBUG   ackrt   << r.ensureConditions   {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}

```

No-Error:
```
1.6455532315793998e+09  DEBUG   ackrt   >> r.ensureConditions   {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}
1.645553231579408e+09   DEBUG   ackrt   >>> rm.IsSynced {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}
1.645553231579413e+09   DEBUG   ackrt   <<< rm.IsSynced {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}
1.645553231579419e+09   DEBUG   ackrt   << r.ensureConditions   {"account": "******", "role": "", "region": "us-west-2", "kind": "API", "namespace": "default", "name": "api-ref-test", "is_adopted": false, "generation": 2}
```